src/modules/blob.js

```diff
@@ -72,15 +74,33 @@ class BlobModule {
    * @return {object} The extrinsic to sign and send.
    */
   createNewTx(blob, keyPair = undefined, signature = undefined) {
+    let value = blob.blob;
+    if (!value) {
+      throw new Error('Blob must have a value!');
+    }
+
+    if (value instanceof Uint8Array) {
+      value = [...value];
+    } else if (typeof value === 'object') {
+      value = stringToHex(JSON.stringify(value));
+    } else if (typeof value === 'string' && value.substr(0, 2) !== '0x') {
```

Comment:
You might end up with invalid hex starting with 0x, like odd length or some non-hex chars, better to use the available utility `isHexWithGivenByteSize` in place of `(typeof value === 'string' && value.substr(0, 2) !== '0x')` and move that check upwards

Reply:
issue with `isHexWithGivenByteSize` is that it requires a byte size, cant blobs be between a min and max length?

Reply:
nvm, i just read docs on `isHexWithGivenByteSize`, notice that byteSize can be undefined - not so clear by naming of function

Reply:
updated, no need to move check above as im trying to exclude whats already hex here